### PR TITLE
Add parsing of jwt to retrieve roles

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ const agenciesRouter = require('./routes/agencies');
 const app = express();
 
 app.use(express.json());
-app.use(express.urlencoded({ extended: false }));
```
Comment (petergphillips) [inline app.js:18]: I just did a format in intellij, not sure which style we prefer?

File: app.js [hunk 2]
```diff
@@ -28,6 +28,9 @@ app.use('/elite2api/persons', personsRouter);
 app.use('/elite2api/images', imagesRouter);
 app.use('/elite2api/offender-relationships', relationshipsRouter);
 app.use('/elite2api/agencies', agenciesRouter);
+app.get('/elite2api/health', (req, res) => {
```
Comment (petergphillips) [inline app.js:31]: Added so that licences app is healthy in docker

File: app.js (hunk 1 above)
Comment (whitfield-mj) [reply app.js:18]: Not hugely concerned for this project really

File: routes/users.js [hunk 3]
```diff
@@ -8,78 +53,31 @@ router.get('/me', function(req, res) {
...
-        ]
-    }
+function translateAuthorities(token, roleHash) {
+  // Authorization expected to be of form 'Bearer x'
+  const accessToken = token.split(' ')[1];
+  try {
+    // try for a real jwt to get the roles from
+    const jwt = jwtDecode(accessToken);
+    const authorities = jwt.authorities;
+    return authorities.map(roleCode => roleHash[roleCode.slice(-2).toLowerCase()]).find(a => a !== null);
```
Comment (whitfield-mj) [inline routes/users.js]: Does it need to check specifically for null? If it is anything falsy then could just do find(a => a)

**Review by whitfield-mj — APPROVED** [review]

**Pull request merged**